src/VirtoCommerce.Platform.Core/JsonConverters/PolymorphJsonConverter.cs

```diff
@@ -4,14 +4,21 @@
 using Newtonsoft.Json.Linq;
 using VirtoCommerce.Platform.Core.Common;
 using VirtoCommerce.Platform.Core.DynamicProperties;
+using VirtoCommerce.Platform.Core.PushNotifications;
 using VirtoCommerce.Platform.Core.Security;
 using VirtoCommerce.Platform.Core.Settings;
 
 namespace VirtoCommerce.Platform.Core.JsonConverters
 {
     public class PolymorphJsonConverter : JsonConverter
     {
-        private static readonly Type[] _knowTypes = { typeof(ObjectSettingEntry), typeof(DynamicProperty), typeof(ApplicationUser), typeof(Role), typeof(PermissionScope) };
+        private static readonly Type[] _knowTypes = { typeof(ObjectSettingEntry), typeof(DynamicProperty), typeof(ApplicationUser), typeof(Role), typeof(PermissionScope), typeof(PushNotification) };
```

Comment:
I think it would be better to reject of using strongly typed `PuschNotification` for inter-process communication, and define the special `DistributedPushNotification` type instead, that will be derived from `PushNotification` and contains a `Dictionary<string, object> Properties` that will be serialized and deserialized from/into Json absolutely transparently for non-strongly-type clients such as virto manager SPA.
https://stackoverflow.com/questions/1207731/how-can-i-deserialize-json-to-a-simple-dictionarystring-string-in-asp-net

Reply:
Built-in type name handling using now